service/common/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -554,14 +554,27 @@ private String resolveNamespaceLocation(Namespace namespace, Map<String, String>
         .map(
             entity -> {
               if (entity.getType().equals(PolarisEntityType.CATALOG)) {
-                return CatalogEntity.of(entity).getDefaultBaseLocation();
+                CatalogEntity catEntity = CatalogEntity.of(entity);
+                String catalogDefaultBaseLocation = catEntity.getDefaultBaseLocation();
+                if (catalogDefaultBaseLocation == null) {
+                  LOGGER.warn(
+                      "Tried to resolve location with catalog with null default base location. Catalog = {}",
+                      catEntity);
+                }
+                return catalogDefaultBaseLocation;
```

Comment:
Should a catalog ever have a null base location? This feels like a case for `PolarisDiagnostics.checkNotNull` rather than a log statement.

Reply:
I'm not too familiar and don't know if this should ever not happen but sounds good to use `PolarisDiagnostics.checkNotNull`. I'm adding some code
```
callContext
              .getPolarisCallContext().getDiagServices().checkNotNull
```
to get the diagnostic services to call checkNotNull though so please check if it's safe to do this or if we should be worried about NPEs.. It seems like elsewhere in the code we're assuming there aren't nulls when calling such methods.